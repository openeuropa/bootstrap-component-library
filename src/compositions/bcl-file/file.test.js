import { renderTwigFileAsNode } from "@openeuropa/bcl-test-utils";
import demoData from "@openeuropa/bcl-file/data.js";
import demoCardData from "@openeuropa/bcl-file/dataCard";
const template = "@oe-bcl/bcl-file/file.html.twig";

const render = (params) => renderTwigFileAsNode(template, params);

describe("OE - File", () => {
  test(`renders correctly`, () => {
    expect.assertions(1);

    return expect(render(demoData)).resolves.toMatchSnapshot();
  });
});

describe("OE - File Card", () => {
  test(`renders correctly`, () => {
    expect.assertions(1);

    return expect(render(demoCardData)).resolves.toMatchSnapshot();
  });
});
