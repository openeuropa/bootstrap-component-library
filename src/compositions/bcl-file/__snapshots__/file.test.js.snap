// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OE - File Card renders correctly 1`] = `
<jest>
  <div
    class="rounded-0 rounded-top p-4 border-bottom-0 card"
  >
    <div
      class="row"
    >
      <div
        class="col-12 col-md-4 order-md-2"
      >
        <img
          alt="alt img"
          class="card-img-top rounded-0 mb-3"
          src="https://picsum.photos/seed/1002/600/400"
        />
      </div>
      <div
        class="col-12 col-md-8"
      >
        <div
          class="card-body p-0"
        >
          <h5
            class="card-title"
          >
            <a
              class="standalone"
              href="#"
            >
              Article title
            </a>
          </h5>
          <p
            class="card-text mb-2"
          >
            Lorem ipsum dolor sit amet, consectetur adipiscing elit. Phasellus ut ex tristique, dignissim sem ac, bibendum est. Sed vehicula lorem non nunc tincidunt hendrerit. Nunc tristique ante et fringilla fermentum.
          </p>
          <span
            class="text-muted d-block d-md-inline-block me-3 mb-3 mb-md-0 text-nowrap"
          >
            Article
          </span>
          <span
            class="text-muted d-block d-md-inline-block me-3 mb-3 mb-md-0 text-nowrap"
          >
            Brussels, Belgium
          </span>
          <span
            class="text-muted d-block d-md-inline-block me-3 text-nowrap"
          >
            17 October 2019
          </span>
          <div
            class="mt-2-5"
          >
            <span
              class="mb-2 me-2 badge rounded-pill badge-outline-primary"
            >
              Label
            </span>
            <span
              class="mb-2 badge rounded-pill badge-outline-primary"
            >
              Label
            </span>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    class="border rounded-bottom"
  >
    <div
      class="px-3-5 py-3"
    >
      <div
        class="d-flex"
      >
        <div
          class="w-100 d-md-flex"
        >
          <div
            class="flex-grow-1"
          >
            <small
              class="fw-bold m-0"
            >
              English
              <span
                class="fw-normal"
              >
                 (16.2 MB - PDF)
              </span>
            </small>
          </div>
          <a
            class="standalone align-self-center d-inline-block mt-1 mt-md-0"
            download="/example.html"
            href="#"
          >
            Download
            <svg
              class="ms-2-5 bi icon--fluid"
            >
              <use
                xlink:href="/icons.svg#download"
              />
            </svg>
          </a>
        </div>
      </div>
    </div>
    <div
      class="bg-lighter py-3 px-4 border-top"
    >
      <div
        class="text-end text-md-center"
      >
        <a
          aria-controls="language-dropdown"
          aria-expanded="false"
          class="standalone"
          data-bs-toggle="collapse"
          href="#language-dropdown"
          role="button"
        >
          Other languages (3)
          <svg
            class="ms-2-5 bi icon--2xs"
          >
            <use
              xlink:href="/icons.svg#caret-down-fill"
            />
          </svg>
        </a>
      </div>
      <div
        class="collapse"
        id="language-dropdown"
      >
        <div
          class="py-3 border-bottom"
        >
          <div
            class="row"
          >
            <div
              class="col-12 col-md-9"
            >
              <p
                class="fw-bold m-0"
              >
                File title FR
              </p>
              <p
                class="m-0"
              >
                Lorem ipsum dolor sit amet, consectetur adipiscing elit. Phasellus ut ex tristique, dignissim sem ac, bibendum est.
              </p>
              <small
                class="fw-bold m-0"
              >
                Français
                <span
                  class="fw-normal"
                >
                   (16.2 MB - PDF)
                </span>
              </small>
            </div>
            <div
              class="col-12 col-md-3 align-self-end text-md-end"
            >
              <a
                class="d-block standalone mt-1"
                download="/example.html"
                href="#"
              >
                Download
                <svg
                  class="ms-2-5 bi icon--fluid"
                >
                  <use
                    xlink:href="/icons.svg#download"
                  />
                </svg>
              </a>
            </div>
          </div>
        </div>
        <div
          class="py-3 border-bottom"
        >
          <div
            class="row"
          >
            <div
              class="col-12 col-md-9"
            >
              <p
                class="fw-bold m-0"
              >
                File title ES
              </p>
              <p
                class="m-0"
              >
                Lorem ipsum dolor sit amet, consectetur adipiscing elit. Phasellus ut ex tristique, dignissim sem ac, bibendum est.
              </p>
              <small
                class="fw-bold m-0"
              >
                Español
                <span
                  class="fw-normal"
                >
                   (16.2 MB - PDF)
                </span>
              </small>
            </div>
            <div
              class="col-12 col-md-3 align-self-end text-md-end"
            >
              <a
                class="d-block standalone mt-1"
                download="/example.html"
                href="#"
              >
                Download
                <svg
                  class="ms-2-5 bi icon--fluid"
                >
                  <use
                    xlink:href="/icons.svg#download"
                  />
                </svg>
              </a>
            </div>
          </div>
        </div>
        <div
          class="py-3 border-bottom"
        >
          <div
            class="row"
          >
            <div
              class="col-12 col-md-9"
            >
              <p
                class="fw-bold m-0"
              >
                File title BG
              </p>
              <p
                class="m-0"
              >
                Lorem ipsum dolor sit amet, consectetur adipiscing elit. Phasellus ut ex tristique, dignissim sem ac, bibendum est.
              </p>
              <small
                class="fw-bold m-0"
              >
                български
                <span
                  class="fw-normal"
                >
                   (16.2 MB - PDF)
                </span>
              </small>
            </div>
            <div
              class="col-12 col-md-3 align-self-end text-md-end"
            >
              <a
                class="d-block standalone mt-1"
                download="/example.html"
                href="#"
              >
                Download
                <svg
                  class="ms-2-5 bi icon--fluid"
                >
                  <use
                    xlink:href="/icons.svg#download"
                  />
                </svg>
              </a>
            </div>
          </div>
        </div>
        <div
          class="pt-3"
        >
          <p
            class="mb-2 d-inline-block me-2"
          >
            Looking for another language which is not on the list? 
          </p>
          <a
            href="/example.html"
          >
            Find out why.
          </a>
        </div>
      </div>
    </div>
  </div>
</jest>
`;

exports[`OE - File Card renders correctly with title 1`] = `
<jest>
  <h6
    class="fw-bold mb-4"
  >
    File card test title
  </h6>
  <div
    class="rounded-0 rounded-top p-4 border-bottom-0 card"
  >
    <div
      class="row"
    >
      <div
        class="col-12 col-md-4 order-md-2"
      >
        <img
          alt="alt img"
          class="card-img-top rounded-0 mb-3"
          src="https://picsum.photos/seed/1002/600/400"
        />
      </div>
      <div
        class="col-12 col-md-8"
      >
        <div
          class="card-body p-0"
        >
          <h5
            class="card-title"
          >
            <a
              class="standalone"
              href="#"
            >
              Article title
            </a>
          </h5>
          <p
            class="card-text mb-2"
          >
            Lorem ipsum dolor sit amet, consectetur adipiscing elit. Phasellus ut ex tristique, dignissim sem ac, bibendum est. Sed vehicula lorem non nunc tincidunt hendrerit. Nunc tristique ante et fringilla fermentum.
          </p>
          <span
            class="text-muted d-block d-md-inline-block me-3 mb-3 mb-md-0 text-nowrap"
          >
            Article
          </span>
          <span
            class="text-muted d-block d-md-inline-block me-3 mb-3 mb-md-0 text-nowrap"
          >
            Brussels, Belgium
          </span>
          <span
            class="text-muted d-block d-md-inline-block me-3 text-nowrap"
          >
            17 October 2019
          </span>
          <div
            class="mt-2-5"
          >
            <span
              class="mb-2 me-2 badge rounded-pill badge-outline-primary"
            >
              Label
            </span>
            <span
              class="mb-2 badge rounded-pill badge-outline-primary"
            >
              Label
            </span>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    class="border rounded-bottom"
  >
    <div
      class="px-3-5 py-3"
    >
      <div
        class="d-flex"
      >
        <div
          class="w-100 d-md-flex"
        >
          <div
            class="flex-grow-1"
          >
            <small
              class="fw-bold m-0"
            >
              English
              <span
                class="fw-normal"
              >
                 (16.2 MB - PDF)
              </span>
            </small>
          </div>
          <a
            class="standalone align-self-center d-inline-block mt-1 mt-md-0"
            download="/example.html"
            href="#"
          >
            Download
            <svg
              class="ms-2-5 bi icon--fluid"
            >
              <use
                xlink:href="/icons.svg#download"
              />
            </svg>
          </a>
        </div>
      </div>
    </div>
    <div
      class="bg-lighter py-3 px-4 border-top"
    >
      <div
        class="text-end text-md-center"
      >
        <a
          aria-controls="language-dropdown"
          aria-expanded="false"
          class="standalone"
          data-bs-toggle="collapse"
          href="#language-dropdown"
          role="button"
        >
          Other languages (3)
          <svg
            class="ms-2-5 bi icon--2xs"
          >
            <use
              xlink:href="/icons.svg#caret-down-fill"
            />
          </svg>
        </a>
      </div>
      <div
        class="collapse"
        id="language-dropdown"
      >
        <div
          class="py-3 border-bottom"
        >
          <div
            class="row"
          >
            <div
              class="col-12 col-md-9"
            >
              <p
                class="fw-bold m-0"
              >
                File title FR
              </p>
              <p
                class="m-0"
              >
                Lorem ipsum dolor sit amet, consectetur adipiscing elit. Phasellus ut ex tristique, dignissim sem ac, bibendum est.
              </p>
              <small
                class="fw-bold m-0"
              >
                Français
                <span
                  class="fw-normal"
                >
                   (16.2 MB - PDF)
                </span>
              </small>
            </div>
            <div
              class="col-12 col-md-3 align-self-end text-md-end"
            >
              <a
                class="d-block standalone mt-1"
                download="/example.html"
                href="#"
              >
                Download
                <svg
                  class="ms-2-5 bi icon--fluid"
                >
                  <use
                    xlink:href="/icons.svg#download"
                  />
                </svg>
              </a>
            </div>
          </div>
        </div>
        <div
          class="py-3 border-bottom"
        >
          <div
            class="row"
          >
            <div
              class="col-12 col-md-9"
            >
              <p
                class="fw-bold m-0"
              >
                File title ES
              </p>
              <p
                class="m-0"
              >
                Lorem ipsum dolor sit amet, consectetur adipiscing elit. Phasellus ut ex tristique, dignissim sem ac, bibendum est.
              </p>
              <small
                class="fw-bold m-0"
              >
                Español
                <span
                  class="fw-normal"
                >
                   (16.2 MB - PDF)
                </span>
              </small>
            </div>
            <div
              class="col-12 col-md-3 align-self-end text-md-end"
            >
              <a
                class="d-block standalone mt-1"
                download="/example.html"
                href="#"
              >
                Download
                <svg
                  class="ms-2-5 bi icon--fluid"
                >
                  <use
                    xlink:href="/icons.svg#download"
                  />
                </svg>
              </a>
            </div>
          </div>
        </div>
        <div
          class="py-3 border-bottom"
        >
          <div
            class="row"
          >
            <div
              class="col-12 col-md-9"
            >
              <p
                class="fw-bold m-0"
              >
                File title BG
              </p>
              <p
                class="m-0"
              >
                Lorem ipsum dolor sit amet, consectetur adipiscing elit. Phasellus ut ex tristique, dignissim sem ac, bibendum est.
              </p>
              <small
                class="fw-bold m-0"
              >
                български
                <span
                  class="fw-normal"
                >
                   (16.2 MB - PDF)
                </span>
              </small>
            </div>
            <div
              class="col-12 col-md-3 align-self-end text-md-end"
            >
              <a
                class="d-block standalone mt-1"
                download="/example.html"
                href="#"
              >
                Download
                <svg
                  class="ms-2-5 bi icon--fluid"
                >
                  <use
                    xlink:href="/icons.svg#download"
                  />
                </svg>
              </a>
            </div>
          </div>
        </div>
        <div
          class="pt-3"
        >
          <p
            class="mb-2 d-inline-block me-2"
          >
            Looking for another language which is not on the list? 
          </p>
          <a
            href="/example.html"
          >
            Find out why.
          </a>
        </div>
      </div>
    </div>
  </div>
</jest>
`;

exports[`OE - File renders correctly 1`] = `
<jest>
  <div
    class="border rounded"
  >
    <div
      class="px-3-5 py-3"
    >
      <div
        class="d-flex"
      >
        <svg
          class="mt-1 icon--file bi icon--xl"
        >
          <use
            xlink:href="/icons.svg#file-pdf-fill"
          />
        </svg>
        <div
          class="w-100 d-md-flex ms-2-5"
        >
          <div
            class="flex-grow-1"
          >
            <p
              class="fw-bold m-0"
            >
              File title
            </p>
            <small
              class="fw-bold m-0"
            >
              English
              <span
                class="fw-normal"
              >
                 (16.2 MB - PDF)
              </span>
            </small>
          </div>
          <a
            class="standalone align-self-center d-inline-block mt-1 mt-md-0"
            download="/example.html"
            href="#"
          >
            Download
            <svg
              class="ms-2-5 bi icon--fluid"
            >
              <use
                xlink:href="/icons.svg#download"
              />
            </svg>
          </a>
        </div>
      </div>
    </div>
    <div
      class="bg-lighter py-3 px-4 border-top"
    >
      <div
        class="text-end text-md-center"
      >
        <a
          aria-controls="language-dropdown"
          aria-expanded="false"
          class="standalone"
          data-bs-toggle="collapse"
          href="#language-dropdown"
          role="button"
        >
          Other languages (3)
          <svg
            class="ms-2-5 bi icon--2xs"
          >
            <use
              xlink:href="/icons.svg#caret-down-fill"
            />
          </svg>
        </a>
      </div>
      <div
        class="collapse"
        id="language-dropdown"
      >
        <div
          class="py-3 border-bottom"
        >
          <div
            class="row"
          >
            <div
              class="col-12 col-md-9"
            >
              <small
                class="fw-bold m-0"
              >
                български
                <span
                  class="fw-normal"
                >
                   (16.2 MB - PDF)
                </span>
              </small>
            </div>
            <div
              class="col-12 col-md-3 align-self-end text-md-end"
            >
              <a
                class="d-block standalone mt-1"
                download="/example.html"
                href="#"
              >
                Download
                <svg
                  class="ms-2-5 bi icon--fluid"
                >
                  <use
                    xlink:href="/icons.svg#download"
                  />
                </svg>
              </a>
            </div>
          </div>
        </div>
        <div
          class="py-3 border-bottom"
        >
          <div
            class="row"
          >
            <div
              class="col-12 col-md-9"
            >
              <small
                class="fw-bold m-0"
              >
                Español
                <span
                  class="fw-normal"
                >
                   (16.2 MB - PDF)
                </span>
              </small>
            </div>
            <div
              class="col-12 col-md-3 align-self-end text-md-end"
            >
              <a
                class="d-block standalone mt-1"
                download="/example.html"
                href="#"
              >
                Download
                <svg
                  class="ms-2-5 bi icon--fluid"
                >
                  <use
                    xlink:href="/icons.svg#download"
                  />
                </svg>
              </a>
            </div>
          </div>
        </div>
        <div
          class="py-3 border-bottom"
        >
          <div
            class="row"
          >
            <div
              class="col-12 col-md-9"
            >
              <small
                class="fw-bold m-0"
              >
                Français
                <span
                  class="fw-normal"
                >
                   (16.2 MB - PDF)
                </span>
              </small>
            </div>
            <div
              class="col-12 col-md-3 align-self-end text-md-end"
            >
              <a
                class="d-block standalone mt-1"
                download="/example.html"
                href="#"
              >
                Download
                <svg
                  class="ms-2-5 bi icon--fluid"
                >
                  <use
                    xlink:href="/icons.svg#download"
                  />
                </svg>
              </a>
            </div>
          </div>
        </div>
        <div
          class="pt-3"
        >
          <p
            class="mb-2"
          >
            Looking for another language which is not on the list? 
          </p>
          <a
            class="d-block"
            href="/example.html"
          >
            Find out why.
          </a>
        </div>
      </div>
    </div>
  </div>
</jest>
`;

exports[`OE - File renders correctly with title 1`] = `
<jest>
  <h6
    class="fw-bold mb-4"
  >
    File test title
  </h6>
  <div
    class="border rounded"
  >
    <div
      class="px-3-5 py-3"
    >
      <div
        class="d-flex"
      >
        <svg
          class="mt-1 icon--file bi icon--xl"
        >
          <use
            xlink:href="/icons.svg#file-pdf-fill"
          />
        </svg>
        <div
          class="w-100 d-md-flex ms-2-5"
        >
          <div
            class="flex-grow-1"
          >
            <p
              class="fw-bold m-0"
            >
              File title
            </p>
            <small
              class="fw-bold m-0"
            >
              English
              <span
                class="fw-normal"
              >
                 (16.2 MB - PDF)
              </span>
            </small>
          </div>
          <a
            class="standalone align-self-center d-inline-block mt-1 mt-md-0"
            download="/example.html"
            href="#"
          >
            Download
            <svg
              class="ms-2-5 bi icon--fluid"
            >
              <use
                xlink:href="/icons.svg#download"
              />
            </svg>
          </a>
        </div>
      </div>
    </div>
    <div
      class="bg-lighter py-3 px-4 border-top"
    >
      <div
        class="text-end text-md-center"
      >
        <a
          aria-controls="language-dropdown"
          aria-expanded="false"
          class="standalone"
          data-bs-toggle="collapse"
          href="#language-dropdown"
          role="button"
        >
          Other languages (3)
          <svg
            class="ms-2-5 bi icon--2xs"
          >
            <use
              xlink:href="/icons.svg#caret-down-fill"
            />
          </svg>
        </a>
      </div>
      <div
        class="collapse"
        id="language-dropdown"
      >
        <div
          class="py-3 border-bottom"
        >
          <div
            class="row"
          >
            <div
              class="col-12 col-md-9"
            >
              <small
                class="fw-bold m-0"
              >
                български
                <span
                  class="fw-normal"
                >
                   (16.2 MB - PDF)
                </span>
              </small>
            </div>
            <div
              class="col-12 col-md-3 align-self-end text-md-end"
            >
              <a
                class="d-block standalone mt-1"
                download="/example.html"
                href="#"
              >
                Download
                <svg
                  class="ms-2-5 bi icon--fluid"
                >
                  <use
                    xlink:href="/icons.svg#download"
                  />
                </svg>
              </a>
            </div>
          </div>
        </div>
        <div
          class="py-3 border-bottom"
        >
          <div
            class="row"
          >
            <div
              class="col-12 col-md-9"
            >
              <small
                class="fw-bold m-0"
              >
                Español
                <span
                  class="fw-normal"
                >
                   (16.2 MB - PDF)
                </span>
              </small>
            </div>
            <div
              class="col-12 col-md-3 align-self-end text-md-end"
            >
              <a
                class="d-block standalone mt-1"
                download="/example.html"
                href="#"
              >
                Download
                <svg
                  class="ms-2-5 bi icon--fluid"
                >
                  <use
                    xlink:href="/icons.svg#download"
                  />
                </svg>
              </a>
            </div>
          </div>
        </div>
        <div
          class="py-3 border-bottom"
        >
          <div
            class="row"
          >
            <div
              class="col-12 col-md-9"
            >
              <small
                class="fw-bold m-0"
              >
                Français
                <span
                  class="fw-normal"
                >
                   (16.2 MB - PDF)
                </span>
              </small>
            </div>
            <div
              class="col-12 col-md-3 align-self-end text-md-end"
            >
              <a
                class="d-block standalone mt-1"
                download="/example.html"
                href="#"
              >
                Download
                <svg
                  class="ms-2-5 bi icon--fluid"
                >
                  <use
                    xlink:href="/icons.svg#download"
                  />
                </svg>
              </a>
            </div>
          </div>
        </div>
        <div
          class="pt-3"
        >
          <p
            class="mb-2"
          >
            Looking for another language which is not on the list? 
          </p>
          <a
            class="d-block"
            href="/example.html"
          >
            Find out why.
          </a>
        </div>
      </div>
    </div>
  </div>
</jest>
`;
